src/publisher.js

```diff
@@ -122,9 +128,37 @@ class Publisher {
   _sendMeasurements(measurements, cb = () => {}) {
     const log = this.registry.logger;
     const uri = this.registry.config.uri;
-    log.debug('Sending ' + measurements.length + ' measurements to ' + uri);
+    log.debug(`Sending ${measurements.length} measurements to ${uri}`);
     const payload = this.payloadForMeasurements(measurements);
-    this.http.postJson(uri, payload, cb);
+    const metricsCallback = (err, res) => {
+      const numMeasurements = measurements.length;
+      if (err) {
+        this.droppedHttp.increment(numMeasurements);
+      } else {
+        let sent = 0;
+        let dropped = 0;
+
+        if (res.statusCode === 200) {
+          sent = numMeasurements;
+        } else if (res.statusCode < 500) {
```

Comment:
Should this be more precise when looking at the status codes? There are others like 408 request timeout that could potentially happen that do not indicate a validation failure. Also in theory a 400 error could have other causes such as posting an invalid JSON payload.